Airport Ticket Booking/Airport Ticket Booking/DBHandler/CsvDataHandler.cs

```diff
@@ -0,0 +1,56 @@
+﻿using CsvHelper;
+using CsvHelper.Configuration;
+using System.Globalization;
+
+namespace AirportTicketBooking.DBHandler
+{
+    internal class CsvDataHandler<TKey, TModel, TMapper> where TKey : IEquatable<TKey> where TMapper : ClassMap<TModel>
+    {
+        protected Dictionary<TKey, TModel> DataDictionary { get; set; } = new();
+        public async Task FetchData(
+            string path, Func<TModel, TKey> extractUnique)
+        {
+            DataDictionary.Clear();
+            var csvConfig = new CsvConfiguration(CultureInfo.InvariantCulture)
+            {
+                HasHeaderRecord = true,
+            };
+            using var reader = new StreamReader(path);
+            using var csv = new CsvReader(reader, csvConfig);
+            csv.Context.RegisterClassMap<TMapper>();
+            var records = csv.GetRecordsAsync<TModel>();
+            await foreach (var record in records)
+            {
+                DataDictionary.Add(extractUnique(record), record);
+            }
+        }
+        public void AppendData(string path, TModel record)
+        {
+            using var stream = File.Open(path, FileMode.Append, FileAccess.Write);
+            using var writer = new StreamWriter(stream);
+            using var csvWriter = new CsvWriter(writer, new CsvConfiguration(CultureInfo.InvariantCulture));
+            csvWriter.WriteRecord(record);
+            csvWriter.NextRecord();
+        }
+        public void AppendDatas(string path, List<TModel> records)
```

Comment:
Datas!  😂 
usually when a method meant to handle batch or bulk of objects,  it's either you call it Bulk<MethodName> or leave it as the one that handle single object and the passed parameter complete the meaning (which I prefer).
`AppendRecord(TModel record)`. => would append one record
`AppendRecord(List<TModel> flights);` would append records  since it take a list.  (don't you think this obvious)